api/src/repositories/submission-repository.ts

```diff
@@ -333,4 +337,39 @@ export class SubmissionRepository extends BaseRepository {
 
     return response.rows[0];
   }
+
+  /**
+   * Fetch a submission record by primary id.
+   *
+   * @param {number} submissionId
+   * @return {*}  {Promise<ISubmissionModel>}
+   * @memberof SubmissionRepository
+   */
+   async listSubmissionRecords(): Promise<ISubmissionModelWithStatus[]> {
+    const sqlStatement = SQL`
+      SELECT
+        t1.submission_status,
+        s.*
+      FROM
+        submission s
+      LEFT JOIN
+        (SELECT DISTINCT ON (ss.submission_id)
+          ss.submission_id,
+          sst.name AS submission_status
+        FROM
+          submission_status ss
+        LEFT JOIN
+          submission_status_type sst
+        ON
+          ss.submission_status_type_id = sst.submission_status_type_id 
+        ORDER BY
+          ss.submission_id, ss.submission_status_id DESC) t1
+      ON
+        t1.submission_id = s.submission_id;
+    `;
+
+    const response = await this.connection.sql<ISubmissionModelWithStatus>(sqlStatement);
+
```

Comment:
Since this method is just listing submissions, it's quite possible that it will return an empty array if there aren't any submissions in the database. Arguably, returning `[]` and not throwing an error is expected behaviour in this case.

Reply:
What about the case where rows doesn't exist on return from sql?

Reply:
In the event that the `submission` table is empty, the expected value of `response.rows` is `[]`. Since this is congruent with what we can expect the method to return in when said table is empty, it's safe to simply return `response.rows`. Are you suggesting that the value of `repsonse.rows` could be something other than `[]` such as `undefined`?

Reply:
Just catching up on this thread.  

I don't believe `rows` can ever be null or undefined, so checking for an array isn't needed. 

`rowCount` could be 0 if there is no status records, but since that is okay, checking for `!response.rowCount` isn't needed.  

So I think this whole if statement can be removed.

Reply:
If something catastrophic occurs it will throw an exception